include/deal.II/grid/manifold.h

```diff
@@ -935,6 +982,33 @@ class ChartManifold : public Manifold<dim,spacedim>
                 const std::vector<double>           &weights) const;
 
   /**
+   * Compute a new set of points around the given points
+   * @p surrounding_points. @p weights is a table with as many columns as
+   * @p surrounding_points.size(). The number of rows in @p weights determines
+   * how many new points will be computed and appended to the last input
+   * argument @p new_points. After exit of this function, the size of
+   * @p new_points equals the size at entry plus the number of rows in
+   * @p weights.
+   *
+   * The implementation of this function first transforms the
+   * @p surrounding_points to the chart by calling pull_back(). Then, new
+   * points are computed on the chart by usual interpolation according to the
+   * given @p weights, which are finally transformed to the image space by
+   * push_forward().
+   *
+   * This implementation can be much more efficient for computing multiple new
+   * points from the same surrounding points than separate calls to
+   * get_new_point() in case the pull_back() operation is expensive. Often,
+   * this is indeed the case because pull_back() might involve Newton
+   * iterations or something similar in non-trivial manifolds.
```

Comment:
I think what you want to say here is that doing multiple points at once helps *reuse* some of the information. You still have to do the pull back for each point separately, it's just that you can (may be able to?) do it more efficiently if you have multiple points at once.